src/sys/socket/mod.rs

```diff
@@ -1934,10 +1934,10 @@ pub fn sockaddr_storage_to_addr(
             Ok(SockAddr::Inet(InetAddr::V6(sin6)))
         }
         libc::AF_UNIX => {
-            let pathlen = len - offset_of!(sockaddr_un, sun_path);
             unsafe {
                 let sun = *(addr as *const _ as *const sockaddr_un);
-                Ok(SockAddr::Unix(UnixAddr::from_raw_parts(sun, pathlen)))
+                let sun_len = len.try_into().unwrap();
```

Comment:
Unwrapping here makes me a little concerned, but I'm not familiar with the previous behavior. Should we return an `Err` instead of panicking?

Reply:
The unwrap is guaranteed to pass, because of the assertion on line 1914.